### ISSUE TITLE
Sonos discovery does not start / not working

### ISSUE THREAD

**Issue by JLindhout** [jishi/node-sonos-http-api#345]
Hi all, 
The program installed perfectly, gives no errors when running (npm start or node server.js) however, it does not discover my Sonos system. (see output at the end)

Some facts / things tried: 
- running in raspberry pi 2
- node v 4.6  / npm v 2.15.9
- Homebridge is succesfully running on node v4.6
- Have added the homebridge-sonos module
- I can ping the IP addresses from Sonos system
- reconfigured the Sonos system (BOOST setup)
- Tried Sonos on my wifi 
- re-installed a few times
- added "disableIpDiscovery": true   to settings.json
- accidently installed  ' sonos ' with npm install sonos  but removed it again 
- when running with trace: 
2016-10-15T13:41:01.944Z TRACE sending M-SEARCH...
2016-10-15T13:41:02.949Z TRACE sending M-SEARCH...
2016-10-15T13:41:03.952Z TRACE sending M-SEARCH...
2016-10-15T13:41:04.954Z TRACE sending M-SEARCH...

Thanks for any further thoughts, am really keen to get this working. 

kind regards Jeroen 

pi@raspberrypi:~/sonos $ node server.js
2016-10-15T13:21:44.450Z INFO Presets loaded: { presets: 
   { players: 
      [ { roomName: 'Woonkamer', volume: 10 },
        { roomName: 'Keuken', volume: 10 },
        { roomName: 'Zolder', volume: 10 } ],
     playMode: { shuffle: true, repeat: 'all', crossfade: false },
     pauseOthers: false,
     favorite: 'My favorite' } }
2016-10-15T13:21:44.634Z INFO http server listening on port 5005
2016-10-15T13:23:06.922Z ERROR No system has yet been discovered. Please see https://github.com/jishi/node-sonos-http-api/issues/77 if it doesn't resolve itself in a few seconds.

**Comment by jishi** [comment]
Does it keep repeating `2016-10-15T13:41:01.944Z TRACE sending M-SEARCH... ` indefinitely when using trace log level? That means the multicast search for a system is failing. Could you please provide the output from `ifconfig`, `route` and `iptables -L`?

**Comment by JLindhout** [comment]
Hi Jimmy, 
Thanks for your fast reply. Indeed " sending M-SEARCH"  is running continously. 

ifconfig eth0
eth0      Link encap:Ethernet  HWaddr b8:27:eb:36:88:2a  
          inet addr:192.168.178.55  Bcast:192.168.178.255  Mask:255.255.255.0
          inet6 addr: 2001:985:5ae2:1:708e:e23f:22d3:95fb/64 Scope:Global
          inet6 addr: fe80::2:6972:94d1:8403/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:11369 errors:0 dropped:6 overruns:0 frame:0
          TX packets:7360 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1529343 (1.4 MiB)  TX bytes:8599368 (8.2 MiB)

route:
route
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         fritz.box       0.0.0.0         UG    202    0        0 eth0
192.168.178.0   *               255.255.255.0   U     202    0        0 eth0

iptables:
sudo iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination       


There is only 1 network adapter on the pi. Hope the above tells you something....

**Comment by jishi** [comment]
How is the pi connected to your network? Have you tried wiring it directly
into one of the players?

**Comment by JLindhout** [comment]
Hi Jimmy, 
The Pi is cables directly to the same router/modem as the main Sonos unit (Connect) is. Tried placing the pi after the CONNECT, but that gives the same result unfortunately. 
Other thoughts? 
regards, Jeroen

**Comment by jishi** [comment]
Your sonos players are also on the 192.168.178.0 network? Does this has a static IP or not?

You have IPv6 configured, which I have never tested with, that might perhaps cause some issues (but probably not). You could try and start the http-api and while it is running (searching) invoke `netstat -natu`  in another shell, that would tell me how the application handles the extra IPv6 stuff.

Lastly, if you have a spare SD-card, you could try and flash my pre-composed image and boot that one to conclude if this is related to your OS-installation or something with the network. However, not if you are running WiFi mode for Sonos, or if all/most players are wired.

**Comment by jishi** [comment]
Just tested activating ipv6 on my raspberry and it doesn't seem to affect anything, so I guess we can rule that out.

Just to be clear, `ifconfig` without eth0 only list `lo` and `eth0` interfaces, correct?

**Comment by JLindhout** [comment]
Hi again, 

Indeed, only eth0 and lo, see below. 

i@raspberrypi:~/sonos $ ifconfig
eth0      Link encap:Ethernet  HWaddr b8:27:eb:36:88:2a  
          inet addr:192.168.178.55  Bcast:192.168.178.255  Mask:255.255.255.0
          inet6 addr: 2001:985:5ae2:1:708e:e23f:22d3:95fb/64 Scope:Global
          inet6 addr: fe80::2:6972:94d1:8403/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:198643 errors:0 dropped:170 overruns:0 frame:0
          TX packets:93735 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:17354659 (16.5 MiB)  TX bytes:64910214 (61.9 MiB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:329890 errors:0 dropped:0 overruns:0 frame:0
          TX packets:329890 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1 
          RX bytes:523318625 (499.0 MiB)  TX bytes:523318625 (499.0 MiB)

pi@raspberrypi:~/sonos $ 

best regards, Jeroen

**Comment by JLindhout** [comment]
Indeed Sonos is on the same network, IP assigned by DHCP, but they are always the same. On top I could fix them in my modem. 

I run the netstat in parallel, output below: 

sudo netstat -natu
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 127.0.0.1:5910          0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:3350          0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:3389            0.0.0.0:*               LISTEN     
tcp       32      0 192.168.178.55:39894    192.30.253.113:443      CLOSE_WAIT 
tcp        0      0 192.168.178.55:3389     192.168.178.26:55166    ESTABLISHED
tcp        0      0 127.0.0.1:5910          127.0.0.1:43350         ESTABLISHED
tcp        0      0 127.0.0.1:46110         127.0.0.1:41812         ESTABLISHED
tcp        0      0 127.0.0.1:43350         127.0.0.1:5910          ESTABLISHED
tcp6       0      0 :::6144                 :::*                    LISTEN     
tcp6       0      0 :::5005                 :::*                    LISTEN     
tcp6       0      0 :::8080                 :::*                    LISTEN     
tcp6       0      0 :::51826                :::*                    LISTEN     
tcp6       0      0 :::41812                :::*                    LISTEN     
tcp6       0      0 :::22                   :::*                    LISTEN     
tcp6       0      0 :::443                  :::*                    LISTEN     
tcp6       0      0 127.0.0.1:41812         127.0.0.1:46110         ESTABLISHED
tcp6       0      0 192.168.178.55:51826    192.168.178.39:58279    ESTABLISHED
udp        0      0 0.0.0.0:68              0.0.0.0:*                          
udp        0      0 0.0.0.0:32870           0.0.0.0:*                          
udp        0      0 192.168.178.55:123      0.0.0.0:*                          
udp        0      0 127.0.0.1:123           0.0.0.0:*                          
udp        0      0 0.0.0.0:123             0.0.0.0:*                          
udp        0      0 0.0.0.0:5353            0.0.0.0:*                          
udp        0      0 0.0.0.0:38241           0.0.0.0:*                          
udp        0      0 0.0.0.0:1905            0.0.0.0:*                          
udp6       0      0 :::46087                :::*                               
udp6       0      0 :::546                  :::*                               
udp6       0      0 fe80::2:6972:94d1:8:123 :::*                               
udp6       0      0 2001:985:5ae2:1:708:123 :::*                               
udp6       0      0 ::1:123                 :::*                               
udp6       0      0 :::123                  :::*                               
udp6       0      0 :::5353                 :::*                               
pi@raspberrypi:~ $ 


I will try your image on a seperate SD card and see what happens. Will report that later.

**Comment by JLindhout** [comment]
Hi again, just burned your image. Interesting results. 

When I put the PI on the router, same results. It does not discover anything. If I put the pi behind the SONOS CONNECT, Going to http://192.168.178.36:5005/zones does give a result. 

[{"uuid":"RINCON_5CAAFD7BCE5801400","coordinator":{"uuid":"RINCON_5CAAFD7BCE5801400","state":{"currentTrack":{"artist":"Sky Radio 101 FM","title":"x-sonosapi-stream:s9067?sid=254&flags=32","albumArtUri":"/getaa?s=1&u=x-sonosapi-stream%3as9067%3fsid%3d254%26flags%3d32","duration":0,"uri":"x-sonosapi-stream:s9067?sid=254&flags=32","type":"radio","absoluteAlbumArtUri":"http://192.168.178.40:1400/getaa?s=1&u=x-sonosapi-stream%3as9067%3fsid%3d254%26flags%3d32"},"nextTrack": ...... ...... ..... 

So looks like something with the network/router. Any clue what setting to change. I am using a FritzBox 7490, latest OS: FRITZ!OS installed:06.52

Best regards, Jeroen

**Comment by jishi** [comment]
Sounds weird that it didn't work with your original installed wired to the CONNECT then. You have no issue with the official controller apps (android, ios, windows, macos) on your network? Are all other controllers running on the wireless network?

It's hard to find info about the fritz!box, especially that aren't in german :) But, according to the manual, you can configure LAN port 4 for guest mode, it would be wise to verify that isn't the case (maybe by trying another port if you used port 4).

In rare occasions, a router can choose to block the SSDP packets (that is used for finding the sonos system) and only respond itselfs. This can usually be mitigated by disabling the UPnP features in the router, with the obvious downside that automatic port forwarding etc will stop working. However, might be worth exploring, and file a bug report if that is the case.

I also saw this quote: "My router Fritz!Box 7490 (from the German company AVM) only works correctly with the outdated Firmware 6.30. Actual Firmwares 6.50 or 6.51 lead to problems as mentioned above. 
Fritz!Box doesn't have any IGMP snooping or multicast forwarding settings." from http://forum.minimserver.com/showthread.php?tid=2722, might be related.

**Comment by JLindhout** [comment]
Update:  I put the raspberry with Sonos image back on the place of the router, left it for a while, and now it also works. So apparently it took some more time to find the Sonos system, but it did. 

So there is something not correctly configured on my original Domoticz image.

**Comment by jishi** [comment]
If you didn't reboot the device when moving over, it would have remembered the discovered players and just continued the communication over the new link. It could still be an issue with SSDP/multicast traffic, since that is only used during startup. The rest of the communication is standard TCP unicast like all other network traffic.

**Comment by jishi** [comment]
This might also relate to your problem: https://frankl.in/code/tweak-fritzbox-for-airplay-across-ethernet-and-wifi but not sure if this is still possible to configure.

**Comment by JLindhout** [comment]
I did shut down the device, and disconnected it from power. Rebooted it again in a different location. 

This now also triggered me to test the original set-up again (on my 2nd pi)........ And surprise, it now also works on that one.... Beats me what happened. 
The only thing I did do in the mean-time was switch off "Media player" on the Fritz! modem. Difficult to test if it is related or not.  

Anyway.... for now it works.... I will do so more test over the coming days, as it is kind of frustrating that it now works all in a sudden. 

Best regards, 
Jeroen
